crates/red_knot/src/symbols.rs

```diff
@@ -413,14 +416,19 @@ impl SymbolTable {
         name: &str,
         flags: SymbolFlags,
     ) -> SymbolId {
-        let hash = SymbolTable::hash_name(name);
         let scope = &mut self.scopes_by_id[scope_id];
-        let name = Name::new(name);
+        let name_id = self
+            .names
+            .get(name)
+            // FIXME: instead of interning here, change `name: &str` arg to `name_id: NameId` and
+            // update callers
```

Comment:
Is it clear this would be an improvement?

Reply:
Yes, I think this current implementation will copy the name every time a symbol with that name is found. Whereas, pushing it up into the caller might save an intern here or there.